test/corpus/autocmds.txt

```diff
@@ -41,17 +41,156 @@ autocmd! BufNewFile *.html echo "foo"
 No arguments
 ================================================================================
 
+au
+autocmd
 au!
 autocmd!
 
 --------------------------------------------------------------------------------
 
 (script_file
+  (autocmd_statement)
+  (autocmd_statement)
   (autocmd_statement
     (bang))
   (autocmd_statement
     (bang)))
 
+================================================================================
+Autocmd only group
+================================================================================
+
+au SingleGroup
+au! SingleGroup
+autocmd single_group
+autocmd! single_group
+
+--------------------------------------------------------------------------------
+
+(script_file
+  (autocmd_statement
+    (augroup_name))
+  (autocmd_statement
+    (bang)
+    (augroup_name))
+  (autocmd_statement
+    (augroup_name))
+  (autocmd_statement
+    (bang)
+    (augroup_name)))
+
+================================================================================
+Autocmd with once and nested
+================================================================================
+
+au BufEnter *.html ++once echo "html or json"
+au! BufEnter *.html ++once echo "html or json"
+au BufEnter *.html ++nested echo "html or json"
+au! BufEnter *.html ++nested echo "html or json"
+
+--------------------------------------------------------------------------------
+
+(script_file
+  (autocmd_statement
+    (au_event_list
+      (au_event))
+    (pattern)
+    (au_once)
+    (echo_statement
+      (string_literal)))
+  (autocmd_statement
+    (bang)
+    (au_event_list
+      (au_event))
+    (pattern)
+    (au_once)
+    (echo_statement
+      (string_literal)))
+  (autocmd_statement
+    (au_event_list
+      (au_event))
+    (pattern)
+    (au_nested)
+    (echo_statement
+      (string_literal)))
+  (autocmd_statement
+    (bang)
+    (au_event_list
+      (au_event))
+    (pattern)
+    (au_nested)
+    (echo_statement
+      (string_literal))))
+
+================================================================================
+Autocmd only event/pattern
+================================================================================
+
+au BufEnter html
+au! BufEnter html
+au BufEnter
+au! BufEnter
+au html
+au! html
+au augroup BufEnter html
+au! augroup BufEnter html
+au augroup BufEnter
+au! augroup BufEnter
+au augroup html
+au! augroup html
+
+--------------------------------------------------------------------------------
+
+(script_file
+  (autocmd_statement
+    (au_event_list
+      (au_event))
+    (pattern))
+  (autocmd_statement
+    (bang)
+    (au_event_list
+      (au_event))
+    (pattern))
+  (autocmd_statement
+    (au_event_list
+      (au_event)))
+  (autocmd_statement
+    (bang)
+    (au_event_list
+      (au_event)))
+  (autocmd_statement
+    (augroup_name))
+  (autocmd_statement
+    (bang)
+    (augroup_name))
```

Comment:
Those should be matched as pattern according to `:h autocmd`, but even vim seems to recognize them as `augroup_name`, as if I type
```vim
autocmd html
```
I get the error `E216: No such group or event: html`, or more appropriate
```vim
autocmd User PackerComplete echo "packer completed"
autocmd PackerComplete
```
get the same error `E216: No such group or event: PackerComplete`